src/App.tsx

```diff
@@ -15,37 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  REVERSE = 'reverse',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [activeSort, setActiveSort] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const resetGoodsOfOrder = () => {
+    setVisibleGoods(goodsFromServer);
+    setActiveSort('');
+  };
+
+  const sortGoodsAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort((a, b) => a.localeCompare(b));
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveSort(SortType.ALPHABET);
+    setIsReversed(false);
+  };
+
+  const sortGoodsByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort((a, b) => a.length - b.length);
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setActiveSort(SortType.LENGTH);
+    setIsReversed(false);
+  }
+
+  const reverseGoodsOrder = () => {
+    setIsReversed(prev => !prev);
+    setVisibleGoods(prev => [...prev].reverse());
+  }
+
```

Comment:
```suggestion
const goods = applySort({ visibleGoods, activeSort, isReversed });
```
Consider creating an `applySort` function that accepts goods and the sorting and returns an array

Reply:
<img width="584" alt="image" src="https://github.com/user-attachments/assets/34d47667-f3b6-4daf-bf73-47451dd20ae7">